hail/python/hailtop/aiocloud/aioaws/fs.py

```diff
@@ -275,11 +276,14 @@ def __str__(self) -> str:
 class S3AsyncFS(AsyncFS):
     schemes: Set[str] = {'s3'}
 
-    def __init__(self, thread_pool: Optional[ThreadPoolExecutor] = None, max_workers: Optional[int] = None):
+    def __init__(self, thread_pool: Optional[ThreadPoolExecutor] = None, max_workers: Optional[int] = None, *, max_pool_connections: int = 10):
```

Comment:
Not for this PR, but it could be worth a discussion on making optional arguments keyword-only before making this API "public"